### PR TITLE
Feature chat

### PR DESCRIPTION
**Pull request by egrzeda**
Finished backend code for chat feature. The POST /relationship/:id/messages route returns the updates status of the relationship, not the added message. This may be changed later, but it likely not necessary.

### CODE REVIEW

File: data/chat.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        if(messageRelationship["mentor"].toString().localeCompare(sender) === 0 || messageRelationship["mentee"].toString().localeCompare(sender) === 0) found = true;
+        if(!found) throw "newMessage: relationship must include sender";
+
+        // Check that message is provided, it is a string, it isn't empty, it isn't just spaces, and it isn't over the maximum message length
+        if(typeof message === "undefined") throw "newMessage: message must be provided";
+        if(typeof message !== "string") throw "newMessage: message must be a string";
+        if(message.length === 0) throw "newMessage: message must not be empty";
+        message = message.trim();
+        if(message.length === 0) throw "newMessage: message must not be just spaces";
+        if(message.length > MAX_MESSAGE_LENGTH) throw "newMessage: message must not be too long";
```
Comment (SaiHarishVitta) [inline data/chat.js:57]: That's an intensive error check.
Can you please move this part to some file under the validations folder?
 That helps in 
1.  distinguishing the business core code from validation code
2. Reusing the validations logic in your client-side js and Routes as well

File: routes/index.js [hunk 2]
```diff
@@ -1,10 +1,13 @@
+
+const chatRoutes = require('./chat')
 const relationships = require("./relationships");
 const postsRoutes = require("./posts");
 
 function constructRoutes(app){
     // Example: app.use('/', <middleware name or router name>); // Mounts router or other middleware at path '/'
     app.use('/posts', postsRoutes);
     app.use('/relationships', relationships);
+    app.use('/relationship', chatRoutes);
```
Comment (SaiHarishVitta) [inline routes/index.js]: We can't have /relationships as one route and /relationship as another route
We can migrate your URLs to /chats if necessary. or we should migrate your functions to relationships.js in routes!

Data level can be different for both but Route should be exposed properly!
We shouldn't confuse clients with /relationships and /relationship

**Review by SaiHarishVitta — CHANGES_REQUESTED** [review]

File: data/chat.js (hunk 1 above)
Comment (SaiHarishVitta) [reply data/chat.js:57]: This will be done in future

File: routes/index.js (hunk 2 above)
Comment (SaiHarishVitta) [reply routes/index.js]: Changed the routes to /chat
And moved getStatus functionality into /relationships.js route
Tested routes and corrected

**Review by daniprogrammi — APPROVED** [review]

**Review by SaiHarishVitta — APPROVED** [review]

**Pull request merged**